From 94a64ec0bc9623244093f1f728cf0fdf99a86cde Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Fri, 28 Feb 2014 12:36:03 -0500
Subject: [PATCH] Revert "KVM: VMX: Auto-load on CPUs with VMX"

This can trigger multiple attempts to load the kvm module,
leading to messages like the following on machines with
32 or more cores:

PERCPU: allocation failed, size=304 align=32, alloc from reserved chunk failed
Call Trace:
Pid: 604, comm: modprobe Tainted: G           O 3.4.79-ovp-ga2-rt95-WR5.0.1.0_preempt-rt #1
Call Trace:
 [<ffffffff8112eb9b>] pcpu_alloc+0x99b/0xa00
 [<ffffffff8112eb9b>] pcpu_alloc+0x99b/0xa00
 [<ffffffff810a02dd>] ? find_symbol_in_section+0x4d/0x140
 [<ffffffff810a02dd>] ? find_symbol_in_section+0x4d/0x140
 [<ffffffff810a0290>] ? find_module+0x70/0x70
 [<ffffffff810a0290>] ? find_module+0x70/0x70
 [<ffffffff810a0e10>] ? each_symbol_section+0x30/0x70
 [<ffffffff810a0e10>] ? each_symbol_section+0x30/0x70
 [<ffffffff810a0e81>] ? find_symbol+0x31/0x60
 [<ffffffff810a0e81>] ? find_symbol+0x31/0x60
 [<ffffffff8112ee93>] __alloc_reserved_percpu+0x13/0x20
 [<ffffffff8112ee93>] __alloc_reserved_percpu+0x13/0x20
 [<ffffffff810a26e1>] load_module+0x3e1/0x1bd0
 [<ffffffff810a26e1>] load_module+0x3e1/0x1bd0
 [<ffffffff811a46f2>] ? fsnotify+0x1d2/0x2b0
 [<ffffffff811a46f2>] ? fsnotify+0x1d2/0x2b0
 [<ffffffff810a3f2b>] sys_init_module+0x5b/0x200
 [<ffffffff810a3f2b>] sys_init_module+0x5b/0x200
 [<ffffffff81823096>] system_call_fastpath+0x1a/0x1f
 [<ffffffff81823096>] system_call_fastpath+0x1a/0x1f
kvm: Could not allocate 304 bytes percpu data

As the mainline solution to this problem is a rather invasive
rewrite of module code to introduce a new state to the state
machine, we simply revert this commit that triggered the auto
load on each core.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 0b8f263db5e8..e68b187704b9 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -27,7 +27,6 @@
 #include <linux/highmem.h>
 #include <linux/sched.h>
 #include <linux/moduleparam.h>
-#include <linux/mod_devicetable.h>
 #include <linux/ftrace_event.h>
 #include <linux/slab.h>
 #include <linux/tboot.h>
@@ -50,12 +49,6 @@
 MODULE_AUTHOR("Qumranet");
 MODULE_LICENSE("GPL");
 
-static const struct x86_cpu_id vmx_cpu_id[] = {
-	X86_FEATURE_MATCH(X86_FEATURE_VMX),
-	{}
-};
-MODULE_DEVICE_TABLE(x86cpu, vmx_cpu_id);
-
 static bool __read_mostly enable_vpid = 1;
 module_param_named(vpid, enable_vpid, bool, 0444);
 
-- 
1.8.5.2

